targets/TARGET_Realtek/TARGET_AMEBA/RTWInterface.cpp

```diff
@@ -0,0 +1,286 @@
+            delete[] accesspoint;
+        }
+        ApNum++;
+    } else{
+        // scan done
+        rtw_up_sema(&scan_sema);
+    }
+    return RTW_SUCCESS;
+}
+
+RTWInterface::RTWInterface()
+    : _dhcp(true), _ip_address(), _netmask(), _gateway()
+{
+    emac_interface_t *emac;
+    int ret;
+
+    emac = wlan_emac_init_interface();
+    if (!emac) {
+       printf("Error init RTWInterface!\r\n");
+    }
+    emac->ops.power_up(emac);
+    ret = mbed_lwip_init(emac);
+    if (ret != 0) {
+        printf("Error init RTWInterface!(%d)\r\n", ret);
+    }
+}
+
+RTWInterface::~RTWInterface()
+{
+    wlan_emac_link_change(false);
+    mbed_lwip_bringdown();
+}
+
+nsapi_error_t RTWInterface::set_network(const char *ip_address, const char *netmask, const char *gateway)
+{
+    _dhcp = false;
+    strncpy(_ip_address, ip_address ? ip_address : "", sizeof(_ip_address));
+    strncpy(_netmask, netmask ? netmask : "", sizeof(_netmask));
+    strncpy(_gateway, gateway ? gateway : "", sizeof(_gateway));
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t RTWInterface::set_dhcp(bool dhcp)
+{
+    _dhcp = dhcp;
+    return NSAPI_ERROR_OK;
+}
+
+/*
+ * we may call connect multiple times
+ */
+nsapi_error_t RTWInterface::set_credentials(const char *ssid, const char *pass, nsapi_security_t security)
+{
+    strncpy(_ssid, ssid, 255);
+    strncpy(_pass, pass, 255);
+    _security = security;
+
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t RTWInterface::connect()
+{
+    int ret;
+    rtw_security_t sec;
+
+    if (!_ssid || (!_pass && _security != NSAPI_SECURITY_NONE)) {
+        printf("Invalid credentials\r\n");
+        return NSAPI_ERROR_PARAMETER;
+    }
+
+    switch (_security) {
+        case NSAPI_SECURITY_WPA:
+        case NSAPI_SECURITY_WPA2:
+        case NSAPI_SECURITY_WPA_WPA2:
+            sec = RTW_SECURITY_WPA_WPA2_MIXED;
+            break;
+        case NSAPI_SECURITY_WEP:
+            sec = RTW_SECURITY_WEP_PSK;
+            break;
+        case NSAPI_SECURITY_NONE:
+            sec = RTW_SECURITY_OPEN;
+            break;			
+        default:
+            return NSAPI_ERROR_PARAMETER;
+    }
+
+    if(_channel > 0 && _channel < 14){
+        uint8_t pscan_config = PSCAN_ENABLE;
+        wifi_set_pscan_chan(&_channel, &pscan_config, 1);
+    }
+    
+    //printf("Connecting to: %s ... \r\n", _ssid);
+    ret = wifi_connect(_ssid, sec, _pass, strlen(_ssid), strlen(_pass), 0, (void *)NULL);
+    if (ret != RTW_SUCCESS) {
+        printf("failed: %d\r\n", ret);
+        return NSAPI_ERROR_NO_CONNECTION;
+    }
+    //printf("connected\r\n");
+
+    wlan_emac_link_change(true);
+    return mbed_lwip_bringup(_dhcp,
+                             _ip_address[0] ? _ip_address : 0,
+                             _netmask[0] ? _netmask : 0,
+                             _gateway[0] ? _gateway : 0);
+}
+
+nsapi_error_t RTWInterface::scan(WiFiAccessPoint *res, unsigned count)
+{
+    static wifi_scan_hdl scan_handler;
+    ApNum = 0;
```

Comment:
Hi @0xc0170 , I move these global variables to class RTWInterface, you might want to check that. Thanks.